src/app/root-store/user-tags/user-tags.reducer.ts

```diff
@@ -0,0 +1,55 @@
+import { UserTagsActionsUnion, UserTagsActionTypes } from './user-tags.actions';
+import { Tag } from '../../shared/models';
+import { initialState, UserTagsState } from './user-tags.state';
+
+
+function getEntities(entities: Tag[]): { [id: string]: Tag } {
+  return entities.reduce((dictionary, entity) => {
+    dictionary[entity.key] = entity;
+    return dictionary;
+  }, {})
+}
+
+function getIds(entities: Tag[]): string[] {
+  return entities.map(tag => tag.key)
+}
+
+function upsertMany(entities: Tag[], state: UserTagsState): UserTagsState {
```

Comment:
Why not to use createEntityAdapter from ngrx/entity??

Reply:
Because it works only with ngrx/entity itself. I did not use it because I can't prepopulate store(ids, entities) during initialization.

Reply:
I was wrong, it's possible. Changed.